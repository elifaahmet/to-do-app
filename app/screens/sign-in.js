import React, { useState } from 'react';
import styled from 'styled-components/native';
import { useDispatch } from 'react-redux';
import { getAuth, signInAnonymously } from "firebase/auth";

import { Label, Input, Button } from '../elements';
import { SafeAreaView } from '../elements';
import { sign_in } from '../redux/userSlice';

const StyledView = styled.View`
    display: flex;
    flex: 1;
    padding-left: 15%;
    padding-right: 15%;
    justify-content: center;
    align-items: center;
    background-color: #E5E5E5;
`;

const StyledElementView = styled.View`
    margin-bottom: 24px;
    width: 100%;
`;

export const SignIn = ({ navigation }) => {

    const [usernameError, setUsernameError] = useState("");
    const [firstNameError, setFirstNameError] = useState("");
    const [username, setUsername] = useState("");
    const [firstName, setFirstName] = useState("");
    const [lastName, setLastName] = useState("");
    const [loading, setLoading] = useState(false);

    const dispatch = useDispatch();

    const handleSignIn = () => {
        if(!username) {
            setUsernameError('Username is required');
            return;
        } else setUsernameError('');

        if(!firstName) {
            setFirstNameError('First Name is required');
            return;
        } else setFirstNameError('');

        setLoading(true);
        const auth = getAuth();
        signInAnonymously(auth)
            .then(() => {
                dispatch(sign_in({
                    username,
                    firstName,
                    lastName
                }));
                setLoading(false);
            })
            .catch((error) => {
                setLoading(false);
                console.log(error.message);
            });
    }


    return (
        <SafeAreaView>
            { loading ? 
            <StyledView>
                <Label>Loading ...</Label>
            </StyledView>
            :
            <StyledView>
                <StyledElementView>
                    <Label>Log in to your account</Label>
                </StyledElementView>
                <StyledElementView>
                    <Input
                        testId="username_input"
                        placeholder="Username"
                        onChangeText={setUsername}
                        value={username}
                        error={usernameError}
                    />
                </StyledElementView>
                <StyledElementView>
                    <Input
                        testId="firstName_input"
                        placeholder="First Name"
                        onChangeText={setFirstName} 
                        value={firstName}
                        error={firstNameError}
                    />
                </StyledElementView>
                <StyledElementView>
                    <Input
                        testId="lastName_input"
                        placeholder="Last Name"
                        onChangeText={setLastName} 
                        value={lastName}
                    />
                </StyledElementView>
                <StyledElementView testId="sign_in">
                    <Button
                        onPress={handleSignIn}
                        title="Sign In"
                    >
                        Sign In
                    </Button>
                </StyledElementView>
            </StyledView>
            }
        </SafeAreaView>
    );
}

export default SignIn;
