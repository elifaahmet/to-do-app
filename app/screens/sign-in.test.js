import * as React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';

import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { SignIn } from './sign-in';
import userReducer from '../redux/userSlice';
import tasksReducer from '../redux/tasksSlice';

const store = configureStore({
    reducer: {
        user: userReducer,
        tasks: tasksReducer
    },
});


describe('Testing Sign In Page', () => {
  test('it should be rendered', async () => {
    const component = (
      <Provider store={store}>
        <SignIn />
      </Provider>
    );

    const {getByPlaceholderText, toJSON} = render(component);

    const element = getByPlaceholderText('Username');
    expect(element.children.length).toEqual(1);

    expect(toJSON()).toMatchSnapshot();
  });

});