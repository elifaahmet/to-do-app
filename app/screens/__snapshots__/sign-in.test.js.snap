// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Sign In Page it should be rendered 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingLeft": "15%",
          "paddingRight": "15%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins_400Regular",
              "fontSize": 18,
              "fontWeight": "600",
              "lineHeight": 28,
              "textAlign": "center",
            },
          ]
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
              "marginBottom": 5,
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          error=""
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Array [
              Object {
                "backgroundColor": "#F1F1F5",
                "borderColor": "#FCFCFC",
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "black",
                "fontFamily": "Poppins_400Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 16,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
          testId="username_input"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
              "marginBottom": 5,
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          error=""
          onChangeText={[Function]}
          placeholder="First Name"
          style={
            Array [
              Object {
                "backgroundColor": "#F1F1F5",
                "borderColor": "#FCFCFC",
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "black",
                "fontFamily": "Poppins_400Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 16,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
          testId="firstName_input"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
              "marginBottom": 5,
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Last Name"
          style={
            Array [
              Object {
                "backgroundColor": "#F1F1F5",
                "borderColor": "#FCFCFC",
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "black",
                "fontFamily": "Poppins_400Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 16,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
          testId="lastName_input"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
            "width": "100%",
          },
        ]
      }
      testId="sign_in"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0062FF",
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FCFCFC",
                "fontFamily": "Poppins_400Regular",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 20,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
